### PR TITLE
Mc murry mess

### PR DESCRIPTION
**Pull request by JustinGOSSES**
---
name: preprocessed McMurray facies dataset
about: requesting the addition of this as a new dataset

This is a dataset with facies and well log curve data from the McMurray & Wabiskaw formations in Alberta, Canada. More information about the [processed dataset's history can be found here](https://github.com/JustinGOSSES/McMurray-Wabiskaw-preprocessed-datasets)
& information about the original dataset [here](https://ags.aer.ca/publications/SPE_006.html): 

---

**Desired dataset/model:**

- Model/dataset name: mcmurray_facies_dataframe.h5.zip 
- Download link: https://github.com/JustinGOSSES/McMurray-Wabiskaw-preprocessed-datasets/raw/master/processed_datasets/mcmurray_facies_dataframe.h5.zip
- License:  Attribution-NonCommercial 2.0 Generic (CC BY-NC 2.0)
- Target audience (geophysicists, solid earth, etc): Geologists, Data Scientists
- Is it widely used: One of the few, if only dataset, open to the public with this many wells (1280) and both facies from core descriptions and well log curves. The S.E.G. facies prediction contest dataset is the most widely used one currently. It only has 3 wells.

-----------

This is to add code for a facies dataset from the McMurray formation in Alberta, Canada. 

I'll do some more work on this pull request to double check format & style as the make format command wasn't able to find Black.

I also need to think about tests

In the requirements.txt I had to add tables as a dependencies as I use a zipped h5 format file and Pandas needs tables to open that type of file.

Fixes #

**Reminders**

- [attempted, need to double check ] Run `make format` and `make check` to make sure the code 

follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ not yet ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ partially complete= ] Write detailed docstrings for all functions/methods.
- [added to examples folder ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

Do not merge this pull request yet. Mostly adding for visibility and to avoid problems when merge is pressed.

### CODE REVIEW

**Comment by JustinGOSSES** [comment]
To load .h5 files into pandas, I need Tables (as it is called in PIP) or PyTables (as it is called in conda) package. In your build pipeline, you load the requirements.txt packages with miniconda. This creates a problem as it tries to load Tables via conda. 

Normally, requirements.txt I think of as for PIP and environment.yml for Conda. What would you suggest doing such that your build passes, but someone can also just run requirements.txt with PIP and not have to always use Conda? Or do you mandate people use Conda?

```
Installing dependencies
===============================================
Capturing dependencies from requirements.txt
Capturing dependencies from requirements-dev.txt
Installing collected dependencies:
pooch>=0.5
xarray
pandas
rasterio
tables
matplotlib
cmocean
cartopy
pytest
pytest-cov
coverage
pylint
flake8
sphinx==1.8.5
sphinx_rtd_theme
sphinx-gallery
numpydoc
twine
codecov
Collecting package metadata: ...working... done
Solving environment: ...working... failed

PackagesNotFoundError: The following packages are not available from current channels:

  - tables

```

**Comment by JustinGOSSES** [comment]
I seem to be blocked by the fact that to load HDFS into Pandas, you need tables/pyTables as a dependency. It is called `tables` in PIP and `pytables` in Conda. 

This is causing a problem as you're install requirements.txt using both pip and conda in the build pipelines.

**Comment by leouieda** [comment]
@JustinGOSSES sorry for the delay. These differences between conda-forge and pypi have caused other headaches in the past. We can't just use the environment.yml file cause it sets the Python version adn we want to test in multiple versions on CI. We can't use pip on CI because of troublesome dependencies. So we're in a bit of a bind.

Currently:

* environment.yml is used to setup a development environment.
* requirements.txt and requirements-dev.txt are used to install conda packages on CI
* setup.py `install_requires` is used to set pip dependencies

This last one is gonna cause problems because I started reading in the requirements from `requirements.txt` to avoid duplication. Clearly this isn't going to work anymore.

I can see 2 ways forward:

1. List pip dependencies explicitly in `setup.py` and conda dependencies in `requirements.txt`
2. Not use hdf5 to store the data and avoid the extra dependency on pytables

So my question is: do we really need to have the data in hdf5? Is it much smaller than xz compressed csv? Is there another binary format that would avoid those extra dependencies?

**Comment by JustinGOSSES** [comment]
Thanks for explanation. I'll try switching it to a xz compressed csv.

**Comment by JustinGOSSES** [comment]
switched, passes checks!

File: rockhound/registry.txt [hunk 1]
```diff
@@ -6,3 +6,4 @@ age.3.2.nc.bz2 49003831db78f7bb044532daf31fdc49218f75876668ee0c96877f383214b827
 age.3.6.nc.bz2 c99358e4a980be15f05df71506d9337dc760d32e5907f9526bc8135224fc1f81 https://www.ngdc.noaa.gov/mgg/ocean_age/data/2008/grids/age/age.3.6.nc.bz2
 ageerror.3.2.nc.bz2 fd599a02488912e6d3bc53fc1e658f3d116a893201c0c9fb5e496c4020279a35 https://www.ngdc.noaa.gov/mgg/ocean_age/data/2008/grids/age_error/ageerror.3.2.nc.bz2
 ageerror.3.6.nc.bz2 15a2542717d30f24feeb2ea9d5d73d754b30a3fc92bbf1b704b3ba23d2766d1b https://www.ngdc.noaa.gov/mgg/ocean_age/data/2008/grids/age_error/ageerror.3.6.nc.bz2
+mcmurray_facies_v1.tar.gz 170315f443096abac9d82d7c5c54b913c8eab183fa8ea112a21db33b4890d05d https://github.com/JustinGOSSES/McMurray-Wabiskaw-preprocessed-datasets/raw/master/processed_datasets/mcmurray_facies_v1.tar.gz
```
Comment (leouieda) [inline rockhound/registry.txt]: It's better to use an exact commit instead of `master` here so that rockhound won't break if changes are made to your repository.

File: rockhound/registry.py [hunk 2]
```diff
@@ -2,7 +2,6 @@
 Create a dataset registry using Pooch and the rockhound/registry.txt file.
 """
 import os
-
```
Comment (leouieda) [inline rockhound/registry.py]: Why are these lines deleted? I imagine it's an automatic editor thing. Could you please add them back?

File: rockhound/mcmurray_facies.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    ##### Insert here all the code to load to take the unzipped files and process into
+    ##### single dataframe for the called dataset
+    ##### These try and excepts are here as some versions of pandas don't work with
+    ##### generic buffers, which fname may be depending on your OS.
+    ##### We get around this by saving as CSV zipped via tar with xz compression.
+    try:
+        test_test = tarfile.open(fname, "r:xz")
+        test_test.extractall("mcmurray_facies_v1")
+        data = pd.read_csv("mcmurray_facies_v1/mcmurray_facies_v1.csv", engine="python")
```
Comment (leouieda) [inline rockhound/mcmurray_facies.py]: Use the `Untar` processor instead: https://www.fatiando.org/pooch/latest/api/generated/pooch.Untar.html#pooch.Untar

File: requirements-dev.txt [hunk 4]
```diff
@@ -6,7 +6,7 @@ pytest-cov
 coverage
 pylint
 flake8
-sphinx=1.8.5
+sphinx==1.8.5
```
Comment (leouieda) [inline requirements-dev.txt:9]: Apparently conda now accepts this. Nice!

File: examples/mcmurray_facies.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+        columns : ['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM', \
+            'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'SitID','lat', \
+                 'lng', 'Depth', 'LithID', 'W_Tar', 'SW', 'VSH', 'PHI', 'RW','lithName']
+"""
+import matplotlib.pyplot as plt
+import rockhound as rh
+
+# Load mcmurray_facies into a DataFrame
+FACIES = rh.fetch_mcmurray_facies()
```
Comment (leouieda) [inline examples/mcmurray_facies.py]:
```suggestion
facies = rh.fetch_mcmurray_facies()
```

File: rockhound/mcmurray_facies.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    Returns
+    -------
+    mcmurray_facies : :class:`pandas.DataFrame` or str
+        The loaded data or the file path to the downloaded data.
+        The :class:`pandas.DataFrame` contains the following data:
+
+        columns : ['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
+        'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'SitID','lat', 'lng',
+         'Depth', 'LithID', 'W_Tar', 'SW', 'VSH', 'PHI', 'RW','lithName']
```
Comment (leouieda) [inline rockhound/mcmurray_facies.py]: I'm not familiar with type of dataset but these names mean nothing to me. What would you think of renaming them to something more verbose? For example, I try to use `latitude` and `longitude` instead of `lat` `lng`/`lon`/`long`. With autocomplete in the notebooks, the length of the names doesn't matter much when typing them out. This can be done when loading the data here so you don't have to change the original files.

**Review by leouieda — CHANGES_REQUESTED** [review]
@JustinGOSSES it feels good to see all the green check marks :heavy_check_mark: Thanks for all the work on this.

There are a few things that need to be done:

* There are some `.DS_Store` files committed that need to be deleted (no need to rewrite history since I'll do a "squash and merge"). Could you also add `.DS_Store` to `.gitignore` so it doesn't happen again in the future?
* Add the `fetch_mcmurry...` function to `doc/api/index.rst`
* Add the Wynne et al citation to `doc/references.rst` and cite it using `[Wynne1995]_` elsewhere (docstring, example, etc)
* See also inline comments

File: rockhound/mcmurray_facies.py (hunk 6 above)
Comment (JustinGOSSES) [reply rockhound/mcmurray_facies.py]: Thanks for all the feedback. 

With exception of 'lat' and 'lng', the abbreviations are standards. I worry about switching to spelled out versions as some people who use this type of data regularly wouldn't then know what curve it was. They only know the abbreviation. 

How about supplying a separate dictionary? Or renaming to long form as default behavior but allowing for abbreviations via an argument submitted to rh.fetch_mcmurray_facies(curveNameAbrev=true)?

Comment (leouieda) [reply rockhound/mcmurray_facies.py]: Oh OK, if it's standard in the field then there is no need to rename. I would recommend including the translation for each abbreviation in the docstring, though. Some people might be looking to get into the field and not know all of the acronyms by heart yet.

File: rockhound/registry.txt (hunk 1 above)
Comment (JustinGOSSES) [reply rockhound/registry.txt]: Changed

File: rockhound/registry.py (hunk 2 above)
Comment (JustinGOSSES) [reply rockhound/registry.py]: Changed

File: rockhound/mcmurray_facies.py (hunk 3 above)
Comment (JustinGOSSES) [reply rockhound/mcmurray_facies.py]: Changed

File: examples/mcmurray_facies.py (hunk 5 above)
Comment (JustinGOSSES) [reply examples/mcmurray_facies.py]: Changed, brings up warning from pylint as doesn't pass naming convention but passes.

File: rockhound/mcmurray_facies.py (hunk 6 above)
Comment (JustinGOSSES) [reply rockhound/mcmurray_facies.py]: Went the abbreviations as default column names but an option to get abbreviation+spelled out column names.

To get the spelled out column names, you're required to add an argument like this: `fetch_mcmurray_facies(abbreviations_only=False)`.  

Default behavior with no arguments `fetch_mcmurray_facies()` is the same thing as calling `fetch_mcmurray_facies(abbreviations_only=True, load=True)`. 

Also included both versions in docstrings.

**Comment by JustinGOSSES** [comment]
Responded to suggestions that are not inline comments:
- Deleted the .DS_Store files and added to .gitignore
- Added the fetch_mcmurray_facies call to doc/api/index.rst
- Added the Wynne et al., citation to the doc/references.rst 

Edited docstrings for clarity for users not familiar with well logs.

**Review by JustinGOSSES — COMMENTED** [review]
All changes made or out of date now.
